apps/accounts/views.py

```diff
@@ -65,6 +65,9 @@ def inner(self, request):
         data = request.GET if request.method == 'GET' else request.DATA
         if 'code' not in data:
             return Response({'error': 'No code provided.'}, status=422)
+        elif ('fxa_state' not in request.session or
+                request.session['fxa_state'] != data['state']):
```

Comment:
did you also mean to check that `decode(data['state'])['user'] == request.user` ? You should check that. Or maybe I am just missing that check?